### DISCUSSION TITLE
Get-Childitem not doing what I want

### DISCUSSION THREAD

**Discussion by Invertible95** [PowerShell/PowerShell#16168] [Q&A] [answered]
Hello there,
I'm fairly new to Powershell and could use some assistance..
I'm trying to get PS to return which person who logged in last on a specific computer. Like this; 
Name              LastWriteTime      
ExampleName 2021-08-15 22:03:49
This is achieved if I run 
`$computer=PCname
Get-ChildItem "\\$computer\c$\Users" | Sort-Object LastWriteTime -Descending | Select-Object Name, LastWriteTime -first 1`
Now I work in a pretty large company and we have named computers, so I compiled them in a .csv file to import instead of doing above script a billion times. And so I came up with the following;
`$computers=Import-Csv 'C:\Temp\LastLogon.csv
foreach ($computer in $computers) {
Get-ChildItem "\\$computer\c$\users" | Sort-Object LastWriteTime -Descending | Select Name, LastWriteTime -first 1
}`
This does not work... When I run it I get back "Get-ChildItem : Cannot find path '\\@{Computer=EG}\c$\Users' because it does not exist." I know the path is correct because it works with the first script. Is the Get-ChildItem not compatible with foreach? Or am I missing something obvious?

First time posting on github, so let me know if this is the wrong place to post or anything else.

Thanks in advance!!

**Comment by vexx32** [comment, marked as answer]
Your CSV import looks like it comes back with objects that have a Computer property you need to pull the computername from before you construct a path with it. For example:

```ps1
$computers = Import-Csv 'C:\Temp\LastLogon.csv'
foreach ($item in $computers) {
    $computer = $item.Computer
    Get-ChildItem -Path "\\$computer\c$\users" |
        Sort-Object -Property LastWriteTime -Descending |
        Select-Object -Property Name, LastWriteTime -First 1
}
```

**Reply by Invertible95** [reply]
Thank you Thank you Thank you!

This worked like a charm!
